pandas/core/internals/blocks.py

```diff
@@ -1401,7 +1410,10 @@ def where(self, other, cond, align=True, errors="raise", try_cast=False, axis=0)
 
         # our where function
         def func(cond, values, other):
-            other = self._try_coerce_args(other)
+
+            if not (self.is_integer and lib.is_scalar(other) and np.isnan(other)):
```

Comment:
seems way too specific here (the is_integer check)

Reply:
I agree, at first had `not self._can_hold_na`, but it turns out _that_ breaks for BoolBlock

Reply:
same issue with the is_integer